front/app/components/FormBuilder/components/FormFields/FormField/Logic/QuestionRuleDisplay.tsx

```diff
@@ -33,40 +39,55 @@ export const QuestionRuleDisplay = ({
 
   return (
     <Box
+      key={key}
       display="flex"
-      ml="52px"
-      height="24px"
       data-cy="e2e-field-rule-display"
+      ml="68px"
+      justifyContent="space-between"
+      alignItems="center"
+      flexWrap="wrap"
+      mb="4px"
     >
-      <Icon
-        fill={isRuleValid ? colors.coolGrey500 : colors.error}
-        width="18px"
-        name="logic"
-        my="auto"
-      />
-      <Text my="auto" pl="8px" pr="4px" color="coolGrey600" fontSize="s">
-        {answer}
-      </Text>
-      <Icon
-        fill={isRuleValid ? colors.teal300 : colors.error}
-        width="16px"
-        name="arrow-right"
-        my="auto"
-      />
-      <Text my="auto" px="4px" color="coolGrey600" fontSize="s">
-        {targetPage}
-      </Text>
-      {!isRuleValid && (
-        <Box my="auto" ml="8px">
-          <Badge
-            style={{ padding: '2px 6px 2px 6px' }}
-            className="inverse"
-            color={colors.error}
-          >
-            <FormattedMessage {...messages.invalidLogicBadgeMessage} />
-          </Badge>
-        </Box>
-      )}
+      <Box flex="3">
+        <Text my="auto" color="coolGrey600" fontSize="s">
+          {answer}
+        </Text>
+      </Box>
+      <Box px="4px">
+        <Icon
+          fill={isRuleValid ? colors.teal300 : colors.error}
+          width="16px"
+          name="arrow-right"
+          my="auto"
+        />
+      </Box>
+      <Box flex="3">
+        <Text
+          my="auto"
+          color="coolGrey600"
+          fontSize="s"
+          display="inline"
+          textDecoration="underline"
+          onClick={(e) => {
+            e.stopPropagation();
+            handleOpenSettings('logic');
+          }}
+        >
+          {targetPage}
+        </Text>
+
+        {!isRuleValid && (
+          <Box my="auto" ml="12px" display="inline">
+            <Badge
+              style={{ padding: '2px 6px 2px 6px' }}
```

Comment:
Minor: we could simplify this to
```suggestion
              style={{ padding: '2px 6px' }}
```